src/components/Icons.tsx

```diff
@@ -44,15 +44,15 @@ interface IconApiStatusProps {
 export const IconTxStatus = ({ status }: IconTxStatusProps) => {
   switch (status) {
     case TransactionStatusEnum.COMPLETED:
-      return <CheckCircleOutlineIcon />;
+      return <CheckCircleOutlineIcon id="check-circle-icon" />;
```

Comment:
```suggestion
      return <CheckCircleOutlineIcon id="test-transaction-status-completed" />;
```
What's the point of bounding this to a concrete component type? 

I'd also distinguish for-tests ids by using `test` prefix everywhere.

Reply:
I use this component to validate in the UI if the step was completed or not.

Reply:
> I use this component to validate in the UI if the step was completed or not.

I understand that. My point is that you are relying on a very specific component to be displayed in something that is a high-level test. The point is that if we decide to change the icon you will need to update the `id` and all references because the id is so tightly coupled with the component being displayed.
Also imagine that we want to write a new test (or assertion). To understand what the test is doing, you need to understand what is `test-check-circle-icon` component and in what circumstances it's display (pretty much the only way to do it so to go to the component and read the code), after doing that you will actually know that: "Aha, that component is displayed when the transaction is completed, so let me use the id.".

So long story short, by using `ids` like that:
1. You are tightly coupling specific design choice (circle icon component) with high-level application logic (transaction is completed). Changing designs will cause extra changes in completely unrelated parts of code.
2. You are creating additional mental overhead to understand what "circle icon component" actually means (transaction is completed).

Reply:
Ok, fair enough ... i will switch it to `test-transaction-step-completed` then